enderio-machines/src/main/java/com/enderio/machines/common/blocks/base/block/MachineBlock.java

```diff
@@ -98,4 +104,27 @@ public void setPlacedBy(Level level, BlockPos pos, BlockState state, @Nullable L
             machine.setMachineOwner(player.getUUID());
         }
     }
+
+    @Override
+    protected ItemInteractionResult useItemOn(ItemStack stack, BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand,
+        BlockHitResult hitResult) {
+        if (!player.getAbilities().instabuild) {
+            return super.useItemOn(stack, state, level, pos, player, hand, hitResult);
+        }
+
+        ISoulBindable soulBindable = level.getCapability(EIOCapabilities.SoulBindable.BLOCK, pos);
```

Comment:
Technically this could be an event so it doesn't have a strict tie to MachineBlock, but I'm happy for now :)